project_name/models/grid_search_model.py

```diff
@@ -0,0 +1,98 @@
+import pandas as pd
+import itertools
+from features.embedding_processing import balance_dataframe, reduce_dimensions
+from models.nn_classifier import create_model, create_optimizer, train_model
+
+
+class GridSearchModel():
+    """
+    A model that performs a grid search over a set of hyperparameters using a neural network
+    """
+
+    hyperparameters = {
```

Comment:
hyperparameters have to be arguments to the function/class. we wouldn't want to go into this class just to change them later

Reply:
you can add assertions when initialized such that you make sure that the necessary parameters exist

Reply:
and probably the place to put it is in the `grid_search.py`, where you actually use the grid search